pandas/tests/indexes/test_base.py

```diff
@@ -2240,6 +2240,30 @@ def test_groupby(self):
 
         tm.assert_dict_equal(result, expected)
 
+    def test_groupby_nan_index_value(self):
+        df = pd.DataFrame([["x", np.nan, 1]], columns=["A", "B", "C"]).set_index(
+            ["A", "B"]
+        )
+        result = df.groupby(level=["A", "B"]).C.sum()
+        result = np.asarray(result)
+
+        s = Series([])
+        s.name = "C"
+        expected = s.astype("int64")
```

Comment:
this test should be much simpler, we don't need to compare the numpy_array_equal, just the resulting Series/DataFrame